contrib/native/client/src/clientlib/zookeeperClient.cpp

```diff
@@ -138,6 +138,11 @@ int ZookeeperClient::getAllDrillbits(const std::string& connectStr, std::vector<
             DRILL_MT_LOG(DRILL_LOG(LOG_TRACE) << "\t Unshuffled Drillbit id: " << drillbits[i] << std::endl;)
         }
     }
+    else{
```

Comment:
I'm not sure the check should be in the zookeeper client, but instead in the drill client: the zookeeper client is just returning the content of the zookeeper tree, and there's no error from the zookeeper client point of view (so no need to disconnect/reconnect).